--- gcc-4.5.3.orig//gcc/config/arm/arm.c        2011-07-14 17:25:05.000000000 +0000
+++ gcc-4.5.3//gcc/config/arm/arm.c     2011-07-14 17:31:26.000000000 +0000
@@ -700,7 +700,7 @@ static int after_arm_reorg = 0;
 /* The maximum number of insns to be used when loading a constant.  */
 static int arm_constant_limit = 3;

-static enum arm_pcs arm_pcs_default;
+enum arm_pcs arm_pcs_default;

 /* For an explanation of these variables, see final_prescan_insn below.  */
 int arm_ccfsm_state;

--- gcc-4.5.3.orig//gcc/config/arm/arm.h.orig	2011-07-21 12:25:47.886316123 +0200
+++ gcc-4.5.3//gcc/config/arm/arm.h	2011-07-21 12:27:44.096316245 +0200
@@ -94,8 +94,14 @@
 	if (arm_arch_iwmmxt)				\
 	  builtin_define ("__IWMMXT__");		\
 	if (TARGET_AAPCS_BASED)				\
-	  builtin_define ("__ARM_EABI__");		\
-    } while (0)
+  {           \
+    if (arm_pcs_default == ARM_PCS_AAPCS_VFP) \
+      builtin_define ("__ARM_PCS_VFP");   \
+    else if (arm_pcs_default == ARM_PCS_AAPCS)  \
+      builtin_define ("__ARM_PCS");   \
+    builtin_define ("__ARM_EABI__");    \
+  }    \
+  } while (0)
 
 /* The various ARM cores.  */
 enum processor_type
@@ -1648,6 +1654,9 @@
   ARM_PCS_UNKNOWN
 };
 
+/* Default procedure calling standard of current compilation unit. */
+extern enum arm_pcs arm_pcs_default;
+
 /* A C type for declaring a variable that is used as the first argument of
    `FUNCTION_ARG' and other related values.  */
 typedef struct
